packages/core/src/browser/keybinding.ts

```diff
@@ -132,60 +131,166 @@ export class KeybindingRegistry {
-                if (collided.length > 0) {
-                    this.logger.warn('Collided keybinding is ignored; ',
-                        Keybinding.stringify(binding), ' collided with ',
-                        collided.map(b => Keybinding.stringify(b)).join(', '));
-                    return;
-                }
-            }
-            if (existingBindings.partial.length > 0) {
-                const shadows = existingBindings.partial.filter(b => b.context === binding.context);
-                if (shadows.length > 0) {
-                    this.logger.warn('Shadowing keybinding is ignored; ',
-                        Keybinding.stringify(binding), ' shadows ',
-                        shadows.map(b => Keybinding.stringify(b)).join(', '));
-                    return;
-                }
-            }
+            const keySequence = KeySequence.parse(this.getPlateformKeybinding(binding.keybinding));
+            result.merge(this.getKeySequenceCollisions(bindings, keySequence));
+        } catch (error) {
+            this.logger.warn(error);
+        }
+        return result;
+    }
+
+    /**
+     * Finds collisions for a key sequence inside a list of bindings (error-free)
+     *
+     * @param bindings the reference bindings
+     * @param keySequence the sequence to match
+     */
+    protected getKeySequenceCollisions(bindings: Keybinding[], keySequence: KeyCode[]): KeybindingRegistry.KeybindingsResult {
+        const result = new KeybindingRegistry.KeybindingsResult();
 
-            if (existingBindings.shadow.length > 0) {
-                const shadows = existingBindings.shadow.filter(b => b.context === binding.context);
-                if (shadows.length > 0) {
-                    this.logger.warn('Shadowed keybinding is ignored; ',
-                        Keybinding.stringify(binding), ' would be shadowed by ',
-                        shadows.map(b => Keybinding.stringify(b)).join(', '));
-                    return;
+        for (const registeredBinding of bindings) {
+            try {
+                const bindingKeySequence = KeySequence.parse(registeredBinding.keybinding);
+                const compareResult = KeySequence.compare(keySequence, bindingKeySequence);
+                switch (compareResult) {
+                    case KeySequence.CompareResult.FULL: {
+                        result.full.push(registeredBinding);
+                        break;
+                    }
+                    case KeySequence.CompareResult.PARTIAL: {
+                        result.partial.push(registeredBinding);
+                        break;
+                    }
+                    case KeySequence.CompareResult.SHADOW: {
+                        result.shadow.push(registeredBinding);
+                        break;
+                    }
+                    default: {
+                        break;
+                    }
                 }
+            } catch (error) {
```

Comment:
Is this try...catch really needed anymore? I don't see which method could throw above.

Reply:
`KeySequence.parse` can throw, even if at this point it shouldn't, the `try/catch` is just to eventually prevent any problem.